x-pack/plugins/enterprise_search/public/applications/app_search/components/source_engines/source_engines_logic.test.ts

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { LogicMounter, mockFlashMessageHelpers, mockHttpValues } from '../../../__mocks__';
+
+jest.mock('../engine', () => ({
+  EngineLogic: { values: { engineName: 'test-engine' } },
+}));
+
+import { nextTick } from '@kbn/test/jest';
+
+import { EngineDetails } from '../engine/types';
+
+import { SourceEnginesLogic } from './source_engines_logic';
+
+const DEFAULT_VALUES = {
+  dataLoading: true,
+  sourceEngines: [],
+};
+
+describe('SourceEnginesLogic', () => {
+  const { http } = mockHttpValues;
+  const { mount } = new LogicMounter(SourceEnginesLogic);
+  const { flashAPIErrors } = mockFlashMessageHelpers;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    mount();
+  });
+
+  it('initializes with default values', () => {
+    expect(SourceEnginesLogic.values).toEqual(DEFAULT_VALUES);
+  });
+
+  describe('setSourceEngines', () => {
+    beforeEach(() => {
+      SourceEnginesLogic.actions.setSourceEngines([
+        { name: 'source-engine-1' },
+        { name: 'source-engine-2' },
+      ] as EngineDetails[]);
+    });
+
+    it('sets the source engines', () => {
+      expect(SourceEnginesLogic.values.sourceEngines).toEqual([
+        { name: 'source-engine-1' },
+        { name: 'source-engine-2' },
+      ]);
+    });
+
+    it('sets dataLoading to false', () => {
+      expect(SourceEnginesLogic.values.dataLoading).toEqual(false);
+    });
+  });
```

Comment:
@JasonStoltz can you weigh in on whether or not you think this is a blocker for logic file tests? We've been writing value checks in most logic files asserting on the entire `values` obj rather than on individual values e.g.:

```js
  describe('setSourceEngines', () => {
    it('sets the source engines & dataLoading to false', () => {
      SourceEnginesLogic.actions.setSourceEngines([
        { name: 'source-engine-1' },
        { name: 'source-engine-2' },
      ] as EngineDetails[]);

      expect(SourceEnginesLogic.values).toEqual({
        ...DEFAULT_VALUES,
        sourceEngines: [
          { name: 'source-engine-1' },
          { name: 'source-engine-2' },
        ],
        dataLoading: false,
      });
    });
  });
```

Personally I'm 50/50 - I do see a large advantage in asserting on the entire `values` being good at catching value/action changes & regressions that devs may forget to write tests for, but I also see the advantage of this way of writing tests (it feels more intuitive and natural to read and I think it works better in terms of dev documentation). I wonder if there's a way to get the best of both worlds with a more complex test helper? 🤷 

My only major horse in this race is that I generally prefer our code base to be consistent across the board, so would like to discuss the approach as a whole if we want to change it!

Reply:
Jason is OOO today so I'm going to go ahead and approve as is rather than block the PR. I'll follow up in Slack for later in case he wants to discuss more in detail or brainstorm a new approach to our logic values tests

Reply:
> Personally I'm 50/50 - I do see a large advantage in asserting on the entire values being good at catching value/action changes & regressions that devs may forget to write tests for, but I also see the advantage of this way of writing tests

I agree. I'm also 50/50. It also gets hairy when you start having selectors that change a lot ... I ended up separating them out from DEFAULT_VALUES so that I could have more targeted tests, and not include selectors when testing state values like you're doing above. https://github.com/elastic/kibana/blob/master/x-pack/plugins/enterprise_search/public/applications/app_search/components/result_settings/result_settings_logic.test.ts#L37-L58

All pros and cons aside, the way we've been doing it has been working so far and I do prefer consistency in any case. I wouldn't call it a blocker but it'd be nice to change this.

Reply:
Oh actually, here's the latest iteration of that convo: https://github.com/elastic/kibana/pull/94769#discussion_r596270306